### PR TITLE
Add 0.4.11 release notes

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR add 0.4.11 release notes. I have not written the "highlights" section - maybe @jni can take a pass on that if he is around, and folks like @kevinyamauchi @alisterburt can contribute pieces too.

Also there are lots of PRs here that I tried to catagorize best I can, any rearrangements are welcome. We probably need to get better at autogenerating them.

Depending on when people can review - as I'm unable to make an rc until release notes are merged - maybe @tlambert03 can give this a quick sanity check now, then we can merge WITHOUT a highlights section - then i can make the rc and folks can be testing in parallel to adding the highlights/ any deprecations/ API changes. I think this will let us move most efficiently through this process. Thanks!!

### CODE REVIEW

File: docs/release/release_0_4_11.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+## Highlights
+
+
+## New Features
+- Add context menu on layer list, introduce `QtActionContextMenu`. (#2556)
+- Add activity dialog and style progress bars (#2656)
+- Add playback options to settings (#2933)
+- Add autoscale modes to image layer model, and buttons to GUI (#3022)
+- Arbitrary plane rendering prototype (#3023)
+- Add view ray and labels selection in 3D (#3037)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md:27]: highlight, obv.

File: docs/release/release_0_4_11.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+- Change opacity slider to float slider (#3016)
+- Add projections to layer context menu, allow grouping and nesting of menu items (#3028)
+- Refactor the Point, Label and Shape Layer Mode logic. (#3050)
+- Make flash effect feel more instant (#3060)
+- Use enum values internally for settings. (#3063)
+- Update vendored volume visual from vispy (#3064)
+- Skip progress indicator test when viewer is not shown (#3065)
+- Allow for multiple installs and update buttons to reflect state (#3067)
+- Unify plugin wording and installer dialog to display only package/plugins (#3071)
+- 3D label picking and label ID in status bar (#3074)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md:65]: feature?  highlight?

File: docs/release/release_0_4_11.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+- Add context menu on layer list, introduce `QtActionContextMenu`. (#2556)
+- Add activity dialog and style progress bars (#2656)
+- Add playback options to settings (#2933)
+- Add autoscale modes to image layer model, and buttons to GUI (#3022)
+- Arbitrary plane rendering prototype (#3023)
+- Add view ray and labels selection in 3D (#3037)
+- Grid mode popup (#3084)
+- Add 3D fill, "mill", and "print" on top of #3074 (#3108)
+- Add positive tail length to tracks layer (#3138)
+- Arbitrary clipping planes for volumes in the image layer (#3140)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md:33]: highlight?

File: docs/release/release_0_4_11.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+
+For more information, examples, and documentation, please visit our website:
+https://github.com/napari/napari
+
+## Highlights
+
+
+## New Features
+- Add context menu on layer list, introduce `QtActionContextMenu`. (#2556)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md:21]: highlight?

File: docs/release/release_0_4_11.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+- Shape Mouse refactor (#2950)
+- Make handling properties more consistent across some layer types (#2957)
+- Labels paintbrush now takes anisotropy into account (#2962)
+- Remove mode guards for selection interactions in points (#2982)
+- Emit data event when moving, adding or removing shapes or points (#2992)
+- Add TypedMutableMapping and EventedDict (#2994)
+- Add isosurface rendering to Labels (#3006)
+- Remove mentions of _mode_history (2987) (#3008)
+- Change opacity slider to float slider (#3016)
+- Add projections to layer context menu, allow grouping and nesting of menu items (#3028)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md]: move to feature?

File: docs/release/release_0_4_11.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+- Use labeled sliders from superqt (#2753)
+- Shortcuts UI (#2864)
+- Convert TextManager to an EventedModel (#2885)
+- Make maximum brush size flexible (#2901)
+- Allow layer to register action on double clicks. (#2907)
+- Reduce numpy array traceback print (#2910)
+- Provide manual deepcopy implementation for translations strings. (#2913)
+- Make Points construction with properties consistent with setting properties (#2916)
+- Add search field to plugin dialog  (#2923)
+- Refactor settings manager to allow setting all preferences with env vars and context managers (#2932)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md]: might want to mention this... though it's mostly a refactor, the "feature" is the ability to now set anything with env vars

File: docs/release/release_0_4_11.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+- Add TypedMutableMapping and EventedDict (#2994)
+- Add isosurface rendering to Labels (#3006)
+- Remove mentions of _mode_history (2987) (#3008)
+- Change opacity slider to float slider (#3016)
+- Add projections to layer context menu, allow grouping and nesting of menu items (#3028)
+- Refactor the Point, Label and Shape Layer Mode logic. (#3050)
+- Make flash effect feel more instant (#3060)
+- Use enum values internally for settings. (#3063)
+- Update vendored volume visual from vispy (#3064)
+- Skip progress indicator test when viewer is not shown (#3065)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md]: move to tests

File: docs/release/release_0_4_11.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+- Allow for multiple installs and update buttons to reflect state (#3067)
+- Unify plugin wording and installer dialog to display only package/plugins (#3071)
+- 3D label picking and label ID in status bar (#3074)
+- Store unmaximized size if napari closes maximized (#3075)
+- Add `add_<shape_type>` method for each shape type (#3076)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md]: feature?

File: docs/release/release_0_4_11.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+- Remove old utils/settings/constants file (#3122)
+- NAPARI_CATCH_ERRORS disable notification manager (#3126)
+- Save files after launch from ipython (#3130)
+- Don't try to read unknown formats in builtin reader plugin (#3145)
+- Make current viewer accessible from the napari module (#3149)
+- Rename `Layer.plane` to `Layer.slicing_plane` (#3150)
+- Update new label action to work with tensorstore arrays (#3153)
+- Always raise PluginErrors (#3157)
+- Establish better YAML encoding for settings (fix enum encoding issue). (#3163)
+- Don't clear blocked plugins when closing preferences dialog (#3164)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md]: bug fix

File: docs/release/release_0_4_11.md [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+## Build Tools, Tests, Documentation, and other Tasks
+- Add imagecodecs to the bundle to open additional tiffs (#2895)
+- Make ordering of releases manual (#2921)
+- Add alister burt to team page (#2937)
+- Use briefcase 0.3.1 on all platforms (#2980)
+- Speedup one of the slowest test. (#2997)
+- Update plugin guide with references and instructions for napari-hub (#3055)
+- Add missing libraries in docker file and entrypoint (#3081)
+- Update documentation regarding the hub (#3091)
+- Fix stubgen and package stubs in wheel/sdists (#3105)
```
Comment (tlambert03) [inline docs/release/release_0_4_11.md]: this is actually a reasonably big quality of life improvement for those using napari in their IDE.  maybe feature?

**Review by tlambert03 — COMMENTED** [review]
wow.  so much!  
left a few thoughts.  will have a look again later

File: docs/release/release_0_4_11.md [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+It's designed for browsing, annotating, and analyzing large multi-dimensional
+images. It's built on top of Qt (for the GUI), vispy (for performant GPU-based
+rendering), and the scientific Python stack (numpy, scipy).
+
+
+For more information, examples, and documentation, please visit our website:
+https://github.com/napari/napari
+
+## Highlights
+
```
Comment (alisterburt) [inline docs/release/release_0_4_11.md:15]:
```suggestion
We've known for a while that 3D interactivity is one of the areas where napari needed improvement (#515). This release introduces ways to use the mouse to interact with data in 3D (#3037). Features like label picking (#3074) and label painting/erasing (#3108) in 3D are just the beginning and we're excited to see where this takes us! For more details, please see the documentation at [https://napari.org/guides/stable/3D_interactivity.html](https://napari.org/guides/stable/3D_interactivity.html).

Our volume rendering functionality has been significantly improved and now includes the ability to render arbitrary planes through volumes (#3023) and add clipping planes to restrict rendering to a region of interest (#3140). For now, these features are marked as `experimental` and the API around their use is likely to change in future versions of napari.

Last but not least, some common operations are now much more accessible from the GUI thanks to a new context menu on the layer list (#2556) and buttons for controlling how autoscaling is performed for images (#3022).
```

**Review by alisterburt — APPROVED** [review]

File: docs/release/release_0_4_11.md [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+https://github.com/napari/napari
+
+## Highlights
+We've known for a while that 3D interactivity is one of the areas where napari needed improvement (#515). This release introduces ways to use the mouse to interact with data in 3D (#3037). Features like label picking (#3074) and label painting/erasing (#3108) in 3D are just the beginning and we're excited to see where this takes us! For more details, please see the documentation at [https://napari.org/guides/stable/3D_interactivity.html](https://napari.org/guides/stable/3D_interactivity.html).
+
+Our volume rendering functionality has been significantly improved and now includes the ability to render arbitrary planes through volumes (#3023) and add clipping planes to restrict rendering to a region of interest (#3140). For now, these features are marked as `experimental` and the API around their use is likely to change in future versions of napari.
+
+Last but not least, some common operations are now much more accessible from the GUI thanks to a new context menu on the layer list (#2556) and buttons for controlling how autoscaling is performed for images (#3022).
```
Comment (alisterburt) [inline docs/release/release_0_4_11.md]:
```suggestion
Last but not least, some common operations are now much more accessible from the GUI thanks to a new context menu on the layer list (#2556 and #3028) and buttons for controlling image contrast limit scaling (#3022).
```

File: docs/release/release_0_4_11.md (hunk 4 above)
Comment (alisterburt) [reply docs/release/release_0_4_11.md:21]: added

File: docs/release/release_0_4_11.md [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+- Shape Mouse refactor (#2950)
+- Make handling properties more consistent across some layer types (#2957)
+- Labels paintbrush now takes anisotropy into account (#2962)
+- Remove mode guards for selection interactions in points (#2982)
+- Emit data event when moving, adding or removing shapes or points (#2992)
+- Add TypedMutableMapping and EventedDict (#2994)
+- Add isosurface rendering to Labels (#3006)
+- Remove mentions of _mode_history (2987) (#3008)
+- Change opacity slider to float slider (#3016)
+- Add projections to layer context menu, allow grouping and nesting of menu items (#3028)
```
Comment (alisterburt) [inline docs/release/release_0_4_11.md]:
```suggestion
```

File: docs/release/release_0_4_11.md [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+- Add playback options to settings (#2933)
+- Add autoscale modes to image layer model, and buttons to GUI (#3022)
+- Arbitrary plane rendering prototype (#3023)
+- Add view ray and labels selection in 3D (#3037)
+- Grid mode popup (#3084)
+- Add 3D fill, "mill", and "print" on top of #3074 (#3108)
+- Add positive tail length to tracks layer (#3138)
+- Arbitrary clipping planes for volumes in the image layer (#3140)
+- Mask image from points layer (#3151)
+
```
Comment (alisterburt) [inline docs/release/release_0_4_11.md]:
```suggestion
- Add projections to layer context menu, allow grouping and nesting of menu items (#3028)
```

**Comment by alisterburt** [comment]
added a first draft of the highlights section, commited directly so that others can add/remove things at will! Thanks for putting this together @sofroniewn

File: docs/release/release_0_4_11.md (hunk 3 above)
Comment (alisterburt) [reply docs/release/release_0_4_11.md:33]: added

File: docs/release/release_0_4_11.md (hunk 2 above)
Comment (alisterburt) [reply docs/release/release_0_4_11.md:65]: added

File: docs/release/release_0_4_11.md (hunk 6 above)
Comment (sofroniewn) [reply docs/release/release_0_4_11.md]: moved to feature

File: docs/release/release_0_4_11.md (hunk 7 above)
Comment (sofroniewn) [reply docs/release/release_0_4_11.md]: done

File: docs/release/release_0_4_11.md (hunk 8 above)
Comment (sofroniewn) [reply docs/release/release_0_4_11.md]: done

File: docs/release/release_0_4_11.md (hunk 9 above)
Comment (sofroniewn) [reply docs/release/release_0_4_11.md]: moved

File: docs/release/release_0_4_11.md (hunk 10 above)
Comment (sofroniewn) [reply docs/release/release_0_4_11.md]: moved

**Comment by sofroniewn** [comment]
Thanks so much @alisterburt for writing the highlights section and going through the release notes - I really like what you wrote and it's a huge help!!

I've rearranged the remaining items flagged by @tlambert03. I think now looks ready to go and will merge after tests pass. I will then make the first rc. We can update release notes during the release process to add things we're missing - @jni

**Pull request merged**